From 8344e0cc25d71267e684bf03d658b7f9d05b2d89 Mon Sep 17 00:00:00 2001
From: Carl Schwan <carl@carlschwan.eu>
Date: Thu, 27 Apr 2023 20:44:05 +0200
Subject: [PATCH] Fix crash trying to access password of empty account

The connect method of SessionPool discard the ImapAccount if it is not
empty the requestPassword method will try to fetch the password for the
empty account and this crash due to a null pointer dereference.

BUG: 435427
---
 resources/imap/sessionpool.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/resources/imap/sessionpool.cpp b/resources/imap/sessionpool.cpp
index 36715767d..c5da6abee 100644
--- a/resources/imap/sessionpool.cpp
+++ b/resources/imap/sessionpool.cpp
@@ -84,7 +84,7 @@ void SessionPool::requestPassword()
 
 bool SessionPool::connect(ImapAccount *account)
 {
-    if (m_account) {
+    if (!account) {
         return false;
     }
 
-- 
2.40.0

